packages/langium/src/serializer/json-serializer.ts

```diff
@@ -6,13 +6,25 @@
 
 import { NameProvider } from '../references/name-provider';
 import { LangiumServices } from '../services';
-import { AstNode, GenericAstNode, isAstNode, isReference, Reference } from '../syntax-tree';
+import { AstNode, CstNode, GenericAstNode, isAstNode, isReference, Reference } from '../syntax-tree';
 import { Mutable } from '../utils/ast-util';
+import { findNodesForProperty } from '../utils/grammar-util';
 import { AstNodeLocator } from '../workspace/ast-node-locator';
+import type { DocumentSegment } from '../workspace/documents';
 
 export interface JsonSerializeOptions {
     space?: string | number
     refText?: boolean
+    textRegions?: boolean
+}
+
+export interface AstNodeWithTextRegions extends AstNode {
+    $textRegion?: AstNodeRegionWithAssignments;
+}
+
+export interface AstNodeRegionWithAssignments extends DocumentSegment {
```

Comment:
What do these assignments and document URI mean?

Reply:
There's documentation missing, right.